File: _cscript/client/gfx/SectorTileSceneNode.cs
smio = global.smoothio

class smio.gfx.SectorTileSceneNode extends CL3D.SceneNode

	constructor: (@engine, @tileNumX, @tileNumY) ->
		super()
		@init()
		#(@mesh = new CL3D.Mesh()).AddMeshBuffer(meshBuf = new CL3D.MeshBuffer())
		#meshBuf.Indices = [0, 1, 3, 1, 2, 3]
		#meshBuf.Vertices.push(@engine.createVertex(@mapWidthHalf, 0, -@mapHeightHalf, 0, 0))
		#meshBuf.Vertices.push(@engine.createVertex(-@mapWidthHalf, 0, -@mapHeightHalf, 1, 0))
		#meshBuf.Vertices.push(@engine.createVertex(-@mapWidthHalf, 0, @mapHeightHalf, 1, 1))
		#meshBuf.Vertices.push(@engine.createVertex(@mapWidthHalf, 0, @mapHeightHalf, 0, 1))
		#meshBuf.Mat.Tex1 = @engine.getTextureManager().getTexture('/_/file/images/textures/earth.png', true)

	OnRegisterSceneNode: (scene) =>
		scene.registerNodeForRendering(@, CL3D.Scene.RENDER_MODE_DEFAULT)
		super(scene)

	render: (renderer) =>
		renderer.setWorld(@getAbsoluteTransformation())
		#renderer.drawMesh(@mesh)

File: default/server/pub/_scripts/gfx/SectorTileSceneNode.js
(function() {
  var smio;
  var __bind = function(fn, me){ return function(){ return fn.apply(me, arguments); }; };
  smio = global.smoothio;
  smio.gfx.SectorTileSceneNode = (function() {
    function SectorTileSceneNode(engine, tileNumX, tileNumY) {
      this.engine = engine;
      this.tileNumX = tileNumX;
      this.tileNumY = tileNumY;
      this.render = __bind(this.render, this);
      this.OnRegisterSceneNode = __bind(this.OnRegisterSceneNode, this);
      SectorTileSceneNode.__super__.constructor.call(this);
      this.init();
    }
    SectorTileSceneNode.prototype.OnRegisterSceneNode = function(scene) {
      scene.registerNodeForRendering(this, CL3D.Scene.RENDER_MODE_DEFAULT);
      return SectorTileSceneNode.__super__.OnRegisterSceneNode.call(this, scene);
    };
    SectorTileSceneNode.prototype.render = function(renderer) {
      return renderer.setWorld(this.getAbsoluteTransformation());
    };
    return SectorTileSceneNode;
  })();
}).call(this);
